flink-runtime/src/main/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorContext.java

```diff
@@ -82,7 +84,7 @@ Licensed to the Apache Software Foundation (ASF) under one
 
     private static final Logger LOG = LoggerFactory.getLogger(SourceCoordinatorContext.class);
 
-    private final ExecutorService coordinatorExecutor;
+    private final ScheduledExecutorService coordinatorExecutor;
```

Comment:
nit: Could we create the `coordinatorExecutor` in the ctor? It's a bit weird now that we close a `coordinatorExecutor` that we are given from the outside of the class. As far as I checked it's always a new service created in the `SourceCoordinatorProvider`.